### ISSUE TITLE
Azure - c7n-mailer Teams integration

### ISSUE THREAD

**Issue by aluong** [cloud-custodian/cloud-custodian#3501]
Add support for Teams with c7n-mailer. We could add explicit support for this or we could add support for an arbitrary webhook and leverage something like Logic Apps that knows how to authenticate with Office 365  to post to Teams.

**Comment by yayitserica** [comment]
### ACCEPTANCE CRITERIA

1. C7N mailer will have a new provider for Teams.

FYI @aluong

**Comment by r32rtb** [comment]
I am just starting to look at Cloud Custodian and hoping to integrate some of these notifications into MS Teams.  Teams uses a webhook that accepts json payload for notifications.  Maybe just a generic webhook module with a templatized message format?

In the case for Teams just a text KV payload works...
curl -H "Content-Type: application/json" -d "{\"text\": \"Hello World\"}" <https://YOUR.TEAMS.INBOUND.WEBHOOK.URL>

**Comment by logachev** [comment]
@r32rtb, yeah you can use generic webhook action for this.. I was thinking about c7n-mailer as a way to send more targeted notifications.
